src/Oscoin/Storage/Block/SQLite.hs

```diff
@@ -37,42 +31,42 @@ import           Codec.Serialise (Serialise)
 -- than starting from scratch. In practice, /for now/, we should be OK as
 -- 'initialize' should be idempotent, but nevertheless we need a better
 -- migration strategy/initialisation here.
-withBlockStore :: ( ToField s
-                  , FromField s
-                  , ToField (Crypto.Hash c)
-                  , FromField (Crypto.Hash c)
-                  , Serialise s
-                  , Serialise (Crypto.Hash c)
-                  , Ord s
-                  , Ord (BlockHash c)
-                  , ToRow tx
-                  , FromRow tx
-                  , Crypto.HasHashing c
-                  )
-               => String
-               -- ^ The path where the DB will live on disk
-               -> Block c tx (Sealed c s)
-               -- ^ The genesis block (used to initialise the store)
-               -> (Block c tx (Sealed c s) -> Score)
-               -- ^ A block scoring function
-               -> (Abstract.BlockStore c tx s IO -> IO b)
-               -- ^ Action which uses the 'BlockStore'.
-               -> IO b
-withBlockStore path genesisBlock score action =
+withBlockStore
+    :: ( ToField s
+       , FromField s
+       , ToField (Crypto.Hash c)
+       , FromField (Crypto.Hash c)
+       , Serialise s
+       , Serialise (Crypto.Hash c)
+       , Ord (BlockHash c)
+       , ToRow tx
+       , FromRow tx
+       , Crypto.HasHashing c
+       )
+    => String
+    -- ^ The path where the DB will live on disk
+    -> Block c tx (Sealed c s)
+    -- ^ The genesis block (used to initialise the store)
+    -> (Abstract.BlockStore c tx s IO -> IO b)
+    -- ^ Action which uses the 'BlockStore'.
+    -> IO b
+withBlockStore path genesisBlock action =
     let newBlockStore internalHandle =
-            ( Abstract.BlockStore {
-                  Abstract.scoreBlock      = hScoreFn internalHandle
-                , Abstract.insertBlock     = storeBlock internalHandle
-                , Abstract.getGenesisBlock = getGenesisBlock (hConn internalHandle)
-                , Abstract.lookupBlock     = lookupBlock internalHandle
-                , Abstract.lookupTx        = lookupTx internalHandle
-                , Abstract.getOrphans      = getOrphans internalHandle
-                , Abstract.getBlocks       = getBlocks internalHandle
-                , Abstract.getTip          = getTip internalHandle
+            ( (Abstract.BlockStoreReader {
+                  Abstract.getGenesisBlock       = getGenesisBlock (hConn internalHandle)
+                , Abstract.lookupBlock           = lookupBlock internalHandle
+                , Abstract.lookupTx              = lookupTx internalHandle
+                , Abstract.getBlocksByDepth      = getBlocks internalHandle
+                , Abstract.getBlocksByParentHash = getChainSuffix (hConn internalHandle)
```

Comment:
Just to make sure: this won't return the block corresponding to the hash, correct? Ie. if you do `getBlocksByParentHash a`, then with the chain `a <- b <- c`, it should only return `b <- c`.

Reply:
I've originally extracted this function from the (already existing) `getChainSuffix` of the SQLite internal module. From what I can see:

```
getChainSuffix conn hsh = do
    result <- lookupBlockTimestamp conn hsh
    case result of
        Nothing -> pure mempty
        Just t -> do
            rows :: [Only (BlockHash c) :. BlockHeader c s] <- Sql.query conn
                [sql|  SELECT hash, parenthash, datahash, statehash, timestamp, difficulty, seal
                         FROM blocks WHERE timestamp > ?
                     ORDER BY timestamp DESC
                        |] (Only t)
            map NF <$> for rows $ \(Only h :. bh) ->
                mkBlock bh <$> getBlockTxs conn h
```

Indeed the behaviour is what you hinted at, as the query is `WHERE timestamp > ?`. 

However, with a fresh mind, I do wonder if this `ByParentHash` suffix ends up being more confusing than anything -- should we relabel this function `getBlocksFromParentHash`? If we want to make this more Java-esque, we can call it `getBlocksFromParentHashExclusive` 😂

Reply:
Ok - I think the name is fine, but I see what you mean. I think `getBlocksWithParentHash` would be definitely wrong, but `by` just means we're using a parent hash to get the blocks, but not what the relationship of the blocks and the hash is.